src/components/safe-apps/AppFrame/useThirdPartyCookies.ts

```diff
@@ -0,0 +1,71 @@
+import { useState, useEffect, useRef, useCallback } from 'react'
+import { THIRD_PARTY_COOKIES_CHECK_URL } from '@/config/constants'
+
+const SHOW_ALERT_TIMEOUT = 10000
+
+const isSafari = (): boolean => {
+  return navigator.userAgent.indexOf('Safari') > -1 && navigator.userAgent.indexOf('Chrome') <= -1
+}
+
+const createIframe = (uri: string, onload: () => void): HTMLIFrameElement => {
+  const iframeElement: HTMLIFrameElement = document.createElement('iframe')
+
+  iframeElement.src = uri
+  iframeElement.setAttribute('style', 'display:none')
+  iframeElement.onload = onload
+
+  return iframeElement
+}
+
+type ThirdPartyCookiesType = {
+  thirdPartyCookiesDisabled: boolean
+  setThirdPartyCookiesDisabled: (value: boolean) => void
+}
+
+const useThirdPartyCookies = (): ThirdPartyCookiesType => {
+  const iframeRef = useRef<HTMLIFrameElement>()
+  const [thirdPartyCookiesDisabled, setThirdPartyCookiesDisabled] = useState<boolean>(false)
+
+  const messageHandler = useCallback((event: MessageEvent) => {
+    const data = event.data
+
+    try {
+      if (data.hasOwnProperty('isCookieEnabled')) {
+        setThirdPartyCookiesDisabled(!data.isCookieEnabled)
+        window.removeEventListener('message', messageHandler)
+        document.body.removeChild(iframeRef.current as Node)
+      }
+    } catch (e) {
+      console.error(e)
```

Comment:
Should we use the logError function here?

Reply:
done!